### PR TITLE
Improved project and item templates

### PR DESCRIPTION
**Pull request by Jaiganeshkumaran**
See #814
- Removed unnecessary content that are basically throw away
- Removed unnecessary spaces
- ViewModel now implements INotifyPropertyChanged (At least that's useful than having a Int32 property)

### CODE REVIEW

**Comment by sylveon** [comment]
If you are making the view model template implement INotifyPropertyChanged, then I would suggest making another template which adds a basic runtime class because I currently use the view model template for that, for lack of better option.

File: vsix/ProjectTemplates/VC/Windows Universal/BlankApp/MainPage.xaml [hunk 1]
```diff
@@ -1,13 +1,13 @@
...
-    mc:Ignorable="d">
+	x:Class="$safeprojectname$.MainPage"
+	xmlns="http://schemas.microsoft.com/winfx/2006/xaml/presentation"
+	xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
+	xmlns:d="http://schemas.microsoft.com/expression/blend/2008"
+	xmlns:local="using:$safeprojectname$"
+	xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006"
+	mc:Ignorable="d">
 
-    <StackPanel Orientation="Horizontal" HorizontalAlignment="Center" VerticalAlignment="Center">
```
Comment (sylveon) [inline vsix/ProjectTemplates/VC/Windows Universal/BlankApp/MainPage.xaml:10]: Keep this for project templates

File: vsix/ProjectTemplates/VC/Windows Universal/StaticLibrary/Class.idl [hunk 2]
```diff
@@ -4,6 +4,5 @@ namespace $safeprojectname$
     runtimeclass Class
     {
         Class();
-        Int32 MyProperty;
```
Comment (sylveon) [inline vsix/ProjectTemplates/VC/Windows Universal/StaticLibrary/Class.idl:7]: Keep this for project templates

File: vsix/ItemTemplates/BlankPage/BlankPage.h [hunk 3]
```diff
@@ -1,17 +1,11 @@
 ﻿#pragma once
-
```
Comment (sylveon) [inline vsix/ItemTemplates/BlankPage/BlankPage.h:2]: This blank line established a stylistic convention in existing C++/WinRT codebases, it should be kept.

File: vsix/ProjectTemplates/VC/Windows Universal/BlankApp/MainPage.xaml [hunk 4]
```diff
@@ -1,13 +1,13 @@
 ﻿<Page
-    x:Class="$safeprojectname$.MainPage"
-    xmlns="http://schemas.microsoft.com/winfx/2006/xaml/presentation"
-    xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
-    xmlns:local="using:$safeprojectname$"
-    xmlns:d="http://schemas.microsoft.com/expression/blend/2008"
-    xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006"
-    mc:Ignorable="d">
+	x:Class="$safeprojectname$.MainPage"
```
Comment (sylveon) [inline vsix/ProjectTemplates/VC/Windows Universal/BlankApp/MainPage.xaml:2]: Use spaces

File: vsix/ProjectTemplates/VC/Windows Universal/BlankApp/App.h [hunk 5]
```diff
@@ -6,7 +6,6 @@ namespace winrt::$safeprojectname$::implementation
     struct App : AppT<App>
     {
         App();
-
```
Comment (sylveon) [inline vsix/ProjectTemplates/VC/Windows Universal/BlankApp/App.h:9]: Keep this line, it's stylistic and meant to visually distinguish the constructor from member methods

File: vsix/ItemTemplates/BlankPage/BlankPage.h (hunk 3 above)
Comment (Jaiganeshkumaran) [reply vsix/ItemTemplates/BlankPage/BlankPage.h:2]: However that isn't consistent with the generated files from IDL file which doesn't have a space after #pragma once

File: vsix/ProjectTemplates/VC/Windows Universal/BlankApp/MainPage.h [hunk 6]
```diff
@@ -1,16 +1,13 @@
 ﻿#pragma once
-
 #include "MainPage.g.h"
 
 namespace winrt::$safeprojectname$::implementation
 {
     struct MainPage : MainPageT<MainPage>
     {
         MainPage();
-
```
Comment (sylveon) [inline vsix/ProjectTemplates/VC/Windows Universal/BlankApp/MainPage.h:10]: Similar to App(), keep this empty line

File: vsix/ProjectTemplates/VC/Windows Universal/WindowsRuntimeComponent/Class.idl [hunk 7]
```diff
@@ -6,4 +6,4 @@ namespace $safeprojectname$
         Class();
         Int32 MyProperty;
     }
-}
+}
```
Comment (sylveon) [inline vsix/ProjectTemplates/VC/Windows Universal/WindowsRuntimeComponent/Class.idl:9]: Keep the newline at the end of the file

File: vsix/ProjectTemplates/VC/Windows Universal/WindowsRuntimeComponent/Class.h [hunk 8]
```diff
@@ -1,13 +1,11 @@
 ﻿#pragma once
-
 #include "Class.g.h"
 
 namespace winrt::$safeprojectname$::implementation
 {
     struct Class : ClassT<Class>
     {
         Class() = default;
-
```
Comment (sylveon) [inline vsix/ProjectTemplates/VC/Windows Universal/WindowsRuntimeComponent/Class.h:10]: Keep the empty line

File: vsix/ItemTemplates/BlankPage/BlankPage.h (hunk 3 above)
Comment (sylveon) [reply vsix/ItemTemplates/BlankPage/BlankPage.h:2]: Fair enough, I haven't noticed

File: vsix/ItemTemplates/ViewModel/ViewModel.h [hunk 9]
```diff
@@ -1,15 +1,16 @@
 ﻿#pragma once
-
+#include "winrt/Windows.UI.Xaml.Data.h"
 #include "$safeitemname$.g.h"
 
 namespace winrt::$rootnamespace$::implementation
 {
     struct $safeitemname$ : $safeitemname$T<$safeitemname$>
     {
         $safeitemname$() = default;
```
Comment (sylveon) [inline vsix/ItemTemplates/ViewModel/ViewModel.h:9]: Add a blank line after constructor, to match the IDL output

File: vsix/ProjectTemplates/VC/Windows Universal/BlankApp/MainPage.h [hunk 10]
```diff
@@ -1,16 +1,14 @@
...
+
```
Comment (sylveon) [inline vsix/ProjectTemplates/VC/Windows Universal/BlankApp/MainPage.h:9]: Remove the spaces, keeping the line fully empty

File: vsix/ProjectTemplates/VC/Windows Universal/StaticLibrary/Class.h [hunk 11]
```diff
@@ -1,13 +1,11 @@
 ﻿#pragma once
-
 #include "Class.g.h"
 
 namespace winrt::$safeprojectname$::implementation
 {
     struct Class : ClassT<Class>
     {
         Class() = default;
-
```
Comment (sylveon) [inline vsix/ProjectTemplates/VC/Windows Universal/StaticLibrary/Class.h:10]: Keep the empty line here as well

**Comment by kennykerr** [comment]
Unfortunately, we have nobody on staff to maintain or test the project templates so they are largely frozen except for critical fixes. Everything under nuget/vsix/natvis is unfortunately very brittle and requires manual testing. Feel free to fork them and provide superior project templates but these will have to stay as they are.

**Pull request closed without merging**